### PR TITLE
Visitors can filter news by categories

### PR DESCRIPTION
**Pull request by johnnysee**
Link to PT: https://www.pivotaltracker.com/story/show/179905397

## User Story

```
As a visitor
In order to find articles relevant to my interests
I want to be able to sort articles by category
```

## Tasks

- Create pop up to contain list of categories
- Add filter functionality for categories
- Clicking a category brings up a page with all relevant articles

## What Have We Learned

- Better understanding in map function overall for example API responses.
- Dropdown functionality for categories with Semantic UI

## Screenshot
A4WbrzTfNO

Screenshot 2021-10-19 at 14 14 28

Screenshot 2021-10-19 at 14 14 37

Screenshot 2021-10-19 at 14 14 48
t

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -8,10 +8,10 @@ import store from "./state/store/configureStore";
 
 window.store = store;
 ReactDOM.render(
-  <React.StrictMode>
+  <>
```
Comment (Fraseer) [inline src/index.js]: Can we get rid of these tags?

File: src/components/HomePage.jsx [hunk 2]
```diff
@@ -2,21 +2,35 @@ import React, { useEffect } from "react";
...
+import { useSelector, useDispatch } from "react-redux";
+import _ from "lodash";
 
 const HomePage = () => {
   const { articles } = useSelector((state) => state);
+  const dispatch = useDispatch();
+
+  const uniqueCategories = () => {
+    let categories = articles.map((article) => {
+      return article.category_name;
```
Comment (Fraseer) [inline src/components/HomePage.jsx:14]: Cool

File: src/components/HomePage.jsx [hunk 3]
```diff
@@ -2,21 +2,35 @@ import React, { useEffect } from "react";
...
+    });
+    dispatch({ type: "SET_CATEGORIES", payload: _.uniq(categories) });
+  };
```
Comment (AugustEnhager) [inline src/components/HomePage.jsx:17]: Looks good with the implementation of lodash

File: src/index.js (hunk 1 above)
Comment (maxarvid) [reply src/index.js]: These were removed to silence the warning about `findDOMNode` being deprecated in `StrictMode`, in response to this morning's post Scrum demo on dropping duplicates from the dynamic render of categories.

Comment (Fraseer) [reply src/index.js]: Sorry I meant getting rid of the two empty tags now, as they dont do much. No complaints about removing StrictMode

Comment (maxarvid) [reply src/index.js]: Hey it works without, good spot. Removing and pushing now.

File: src/modules/apiHelper.js [hunk 4]
```diff
@@ -2,9 +2,19 @@ import axios from "axios";
...
+      if (categoryName) {
+        response = await axios.get(
+          "https://news-in-progress-api.herokuapp.com/api/articles/",
+          { params: { category_name: categoryName } }
+        );
+      } else {
+        response = await axios.get(
+          "https://news-in-progress-api.herokuapp.com/api/articles/"
+        );
+      }
```
Comment (tochman) [inline src/modules/apiHelper.js]:
```js
const options = categoryName ? { category_name: categoryName } : {}
```

Another version
```js
const options = categoryName && { category_name: categoryName }
```


```js
response = await axios.get(
          "https://news-in-progress-api.herokuapp.com/api/articles/",
          { params: options }
        );
```

Comment (dorianbuck) [reply src/modules/apiHelper.js]: This has been applied

**Review by tochman — APPROVED** [review]
Looks great!

**Review by maxarvid — APPROVED** [review]
LGTM

File: src/styles/index.css [hunk 5]
```diff
@@ -1,4 +1,9 @@
-
+h1 {
+  font-size: 3em;
+  font-family:'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; 
+  color: black;
+  text-align: center;
+}
```
Comment (tochman) [inline src/styles/index.css:6]:
```css
h1, h1, h3 {
 font-family:'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}
```

**Pull request merged**